bip-0000.mediawiki

```diff
@@ -0,0 +1,222 @@
+
+==== Scanning ====
+
+For each transaction the receiving wallet suspects might be a silent payment to themselves, it must<ref name="data_needed_for_scanning>'''What data does a wallet need for scanning''' A wallet needs a 32 byte public key which represents the sum of the input public keys of a transaction, the truncated 8 byte hash of the transaction input outpoints, and some method of determining whether or not a scriptPubKey exists in a block. For full nodes, all of this data is already available. For light clients, they would need to request the tweak data from a server with access to the necessary information for tweaking and determine existence in a block using something like BIP158. BIP158 is recommended over other methods as this is the most trust minimized, privacy preserving method for light clients today.</ref>:
+
+* Generate the ''outpoint_hash'', using the method described above
+* Let ''A = A<sub>0</sub> + A<sub>1</sub> + … A<sub>i</sub>'', where each ''A<sub>i</sub>'' is the public key corresponding to the ''ith'' input of the transaction
+* Compute the tweaks:
+** Let ''M = the number of labels the Recipient wallet has used''
+** Let ''t<sub>0</sub> = SHA256(SHA256(outpoint_hash·b<sub>scan</sub>·A || 0))''
+** Compute ''P<sub>00</sub> = t<sub>0</sub>·G + B<sub>spend</sub>''
+** If ''M'' is greater than 0:
+*** Let ''m = 1'':
+*** While ''m'' &lt;= ''M'':
+**** Compute ''P<sub>m0</sub> = P<sub>00</sub> + B<sub>spend</sub> + m·G''
+**** repeat with ''m++''
+** For each public key ''P<sub>m0</sub>'' in {''P<sub>00</sub> … P<sub>M0</sub>''}:
+*** If ''P<sub>m0</sub>'' is detected in the transaction outputs, starting with ''n = 1'':
+**** Compute ''t<sub>n</sub> = SHA256(SHA256(outpoint_hash·b<sub>scan</sub>·A || n))''
+**** Check if ''P<sub>mn</sub> = (t<sub>n</sub> + m)·G + B<sub>spend</sub>'' is present in the remaining outputs
+**** If not, terminate
+**** If ''P<sub>mn</sub>'' is found, repeat with ''n++''
+
+For each output with public key ''P<sub>mn</sub>'', store the output, label ''m'', and tweak data ''t<sub>n</sub>'' in the wallet. ''P<sub>mn</sub>'' is spendable with the private key ''(x = t<sub>n</sub> + m + b<sub>spend</sub>) mod p''
+
+==== Backup and Recovery ====
+
+Since each silent payment output address is derived independently, descriptor wallets are recommended along with regular backups. When recovering from a backup, the wallet will need to scan since the last backup to detect new payments.
+
+If using a seed phrase only style backup, the user can recover the wallet’s current state from the UTXO set and can recover the wallet history by scanning the blockchain, starting from the wallet birthday.
+
+===== Recovery from the UTXO set =====
```

Comment:
The “UTXO set” are all UTXOs in existence globally. I like to use “UTXO pool” to refer to the UTXOs owned by one wallet.

```suggestion
===== Recovery from the UTXO pool =====
```

Reply:
Oh d’uh. For some reason I read that as “Recover of the UTXO set”